v2/uci/src/modules/adapter/adapter.service.spec.ts

```diff
@@ -1,18 +1,142 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { AdaptersService } from './adapter.service';
+import { PrismaService } from '../../global-services/prisma.service';
+import { createMock } from '@golevelup/ts-jest';
+import { TelemetryService } from '../../global-services/telemetry.service';
 
 describe('AdaptersService', () => {
   let service: AdaptersService;
+  let mockPrismaService: PrismaService;
+
+  const mockAdapter = {
+    id: 'mockID',
+    createdAt: 1000000000000,
+    updatedAt: 1000000000000,
+    channel: 'mockChannel',
+    provider: 'mockProvider',
+    config: {
+      phone: '0000000000',
+      HSM_ID: 'mockHSMID',
+      '2WAY': 'mock2WAY',
+      credentials: { ['mockKey']: 'mockKeys' },
+    },
+    name: 'mockName',
+  };
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [AdaptersService],
-    }).compile();
+      providers: [
+        {
+          provide: PrismaService,
+          useValue: {
+            adapter: {
+              create: jest.fn(async (adapterDTO) => {
+                return {
+                  id: 'mockID',
+                  createdAt: 1000000000000,
+                  updatedAt: 1000000000000,
+                  channel: adapterDTO.data.channel,
+                  provider: adapterDTO.data.provider,
+                  config: adapterDTO.data.config,
+                  name: adapterDTO.data.name,
+                };
+              }),
+              findMany: jest.fn().mockReturnValue([
+                {
+                  ...mockAdapter,
+                },
+              ]),
+              findUnique: jest.fn((filter) => {
+                const res = { ...mockAdapter };
+                res.id = filter.where.id;
+                return res;
+              }),
+              update: jest.fn((payload) => {
+                return {
+                  id: payload.where.id,
+                  createdAt: 1000000000000,
+                  updatedAt: 1000000000000,
+                  ...payload.data,
+                };
+              }),
+            },
```

Comment:
Can you extract the mock implementation above `beforeEach`? Looks a bit cleaner that way, as this mock implementation might increase in size later, as needed.

Reply:
Sure, can you review the recent changes. Thanks!